com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/EvaluateRequestHandler.java

```diff
@@ -64,6 +64,11 @@ public CompletableFuture<Response> handle(Command command, Arguments arguments,
         VariableUtils.applyFormatterOptions(options, evalArguments.format != null && evalArguments.format.hex);
         String expression = evalArguments.expression;
 
+        // Async mode is supposed to be performant, then disable the advanced features like hover evaluation.
+        if (context.isAttached() && context.asyncJDWP() && "hover".equals(evalArguments.context)) {
```

Comment:
Maybe advanced features can still be turned on if it's attach mode but the host is local?

Reply:
Good suggestion.